charts/kong/ci/test5-values.yaml

```diff
@@ -0,0 +1,43 @@
+# This tests the following unrelated aspects of Ingress Controller
+# - ingressController deploys with a database
+# - TODO database-backed proxy works _without_ stream listens
+#   This test is a duplicate of test2-values.yaml to handle the https://github.com/Kong/charts/issues/295
+#   workaround when the socket file is _not_ created
```

Comment:
I don't understand this comment.
Does `test5-values.yaml` (identical to `test2-values.yaml`) verify that `test2` has not leaked the socket by retrying the same operations and failing if the socket already exists?

If so:
```suggestion
#   This test is a duplicate of test2-values.yaml to handle the https://github.com/Kong/charts/issues/295
#   in order to check if test2 hasn't leaked a socket file. If it did, this test
#   will fail because the file already exists.
```

Reply:
Rephrased to clarify the TODO condition and purpose of the test.

This new case was more necessary when the workaround used a conditional test, before your suggestion to use `rm -f`. With the prior iteration, I wanted to test both workaround cases, i.e.
- A DB-backed configuration with a stream listen creates a socket file in wait-for-db. The workaround removes it, wait-for-db exits successfully, and the main Kong container starts correctly. test2 already confirmed this.
- A DB-backed configuration with _no_ stream listen creates no socket files in wait-for-db. The workaround does not fail if no file is present, wait-for-db exits successfully, and the main Kong container starts correctly. No existing test case covered this; test5 now does.

test2 and test5 don't interact with one another, but we can't simultaneously test the workaround behavior for configurations that create the file and those that do not, and no existing test covered the latter.

The workaround is arguably safer with the `rm -f` variant (we rely on known behavior of `rm` rather than our own script), so we may be okay to remove it entirely--I'm ambivalent, but keeping it errs on the side of caution. Comment is hopefully clearer now though.